### PR TITLE
Integrated ledger search by id

### PR DESCRIPTION
**Pull request by piercarlobeghetto**
Is now possible in ledger's view filter by ledger ids.
The filter works with multiple ids (only long values separated by comma). 
If any of the provided ids isn't parsable, the filter provides no results.

### CODE REVIEW

File: web/src/main/java/org/bookkeepervisualmanager/api/resources/LedgersResource.java [hunk 1]
```diff
@@ -101,4 +101,30 @@ private Integer convertParam(String s) {
         }
     }
 
+    /**
+     * Split a string into a Long list. If any value is invalid, return empty list
+     *
+     * @param s
+     * @return List
+     */
+    private List<Long> convertParamLongList(String s) {
```
Comment (eolivelli) [inline web/src/main/java/org/bookkeepervisualmanager/api/resources/LedgersResource.java]: static

File: web/src/main/java/org/bookkeepervisualmanager/api/resources/LedgersResource.java [hunk 2]
```diff
@@ -101,4 +101,30 @@ private Integer convertParam(String s) {
...
+        if (s == null || s.trim().isEmpty()) {
+            return null;
```
Comment (eolivelli) [inline web/src/main/java/org/bookkeepervisualmanager/api/resources/LedgersResource.java]: it is a code bad practice to return sometimes "null" and sometimes not.
Please return an empty list

File: web/src/main/java/org/bookkeepervisualmanager/api/resources/LedgersResource.java [hunk 3]
```diff
@@ -101,4 +101,30 @@ private Integer convertParam(String s) {
...
+        }
+        List<Long> res = new ArrayList<>();
+        for (String _s : s.split(",")) {
+            _s = _s.trim();
+            if (!_s.isEmpty()) {
+                try {
+                    Long l = Long.parseLong(_s);
+                    res.add(l);
+                } catch (NumberFormatException ex) {
+                    res.clear();
```
Comment (eolivelli) [inline web/src/main/java/org/bookkeepervisualmanager/api/resources/LedgersResource.java]: in this case we won't filter out any id.
Proposals:
- ignore the value
- throw an error (and handle it upstream)

I am happy with any of the two choices

File: web/src/test/java/org/bookkeepervisualmanager/cache/MetadataCacheTest.java [hunk 4]
```diff
@@ -49,12 +50,16 @@ public void test() {
...
-                List<Ledger> ledgersByMeta = metadataCache.searchLedgers("foo", null);
+                List<Ledger> ledgersByMeta = metadataCache.searchLedgers("foo", null, null);
                 assertEquals(1, ledgersByMeta.size());
-                List<Ledger> ledgersByBookie = metadataCache.searchLedgers(null, "localhost:1234");
+                List<Ledger> ledgersByBookie = metadataCache.searchLedgers(null, "localhost:1234", null);
                 assertEquals(1, ledgersByBookie.size());
-                List<Ledger> ledgersByBookieAndMeta = metadataCache.searchLedgers("pulsar", "localhost:1234");
+                List<Ledger> ledgersByBookieAndMeta = metadataCache.searchLedgers("pulsar", "localhost:1234", null);
                 assertEquals(1, ledgersByBookieAndMeta.size());
+                List<Ledger> ledgersByBookieAndMetaAndIdOk = metadataCache.searchLedgers("pulsar", "localhost:1234", Arrays.asList(new Long[] {1L}));
```
Comment (eolivelli) [inline web/src/test/java/org/bookkeepervisualmanager/cache/MetadataCacheTest.java]: `Arrays.asList(1L)`

File: web/src/test/java/org/bookkeepervisualmanager/cache/MetadataCacheTest.java [hunk 5]
```diff
@@ -49,12 +50,16 @@ public void test() {
...
+                assertEquals(1, ledgersByBookieAndMetaAndIdOk.size());
+                List<Ledger> ledgersByBookieAndMetaAndIdKo = metadataCache.searchLedgers("pulsar", "localhost:1234", new ArrayList());
```
Comment (eolivelli) [inline web/src/test/java/org/bookkeepervisualmanager/cache/MetadataCacheTest.java]: Collections.emptyList ?
please also some negative tests, like looking for ledger ids that do not exist
and mix existing/non existant ledger ids

File: web/src/main/java/org/bookkeepervisualmanager/api/resources/LedgersResource.java (hunk 3 above)
Comment (piercarlobeghetto) [reply web/src/main/java/org/bookkeepervisualmanager/api/resources/LedgersResource.java]: Method now throws a NumberFormatException handled to check errors.

**Pull request merged**